locations/models.py

```diff
@@ -186,8 +184,7 @@ def get_context(self, request):
 
         # Filter by tag
         tag = request.GET.get('tag')
-        locationsindexset= LocationsIndexPage.objects.live().parent_of(self)
-        locationpages = LocationPage.objects.filter(tags__name=tag).live().descendant_of(locationsindexset[0])
+        locationpages = LocationPage.objects.filter(tags__name=tag).live().descendant_of(self.get_parent())
```

Comment:
maybe a comment would be good: `# the location pages which are descendants of the parent page`